### PR TITLE
Add option to ignore NaNs in biweight stats

### PR DESCRIPTION
**Pull request by larrybradley**
This PR adds an `ignore_nan` keyword option to the `biweight_location`, `biweight_scale`, and `biweight_midvariance` functions.  The `ignore_nan` keyword already exists in `median_absolute_deviation` and `mad_std`.

### CODE REVIEW

**Comment by bsipocz** [comment]
can we start making these keyword only args now, or would rather wait for a discussion about it at the coordination meeting?

**Comment by larrybradley** [comment]
I'm fine with changing it now.  Do we have a standard?  I wasn't aware of one.

**Comment by pllim** [comment]
> Do we have a standard? 

Not yet but looks like we're moving that way, especially after the coordination meeting. Might as well start now? :wink:

**Comment by larrybradley** [comment]
I've made `ignore_nan` be a keyword-only argument.

File: astropy/stats/biweight.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
            'biweight_midcovariance', 'biweight_midcorrelation']
 
 
-def biweight_location(data, c=6.0, M=None, axis=None):
+def biweight_location(data, c=6.0, M=None, axis=None, *, ignore_nan=False):
```
Comment (pllim) [inline astropy/stats/biweight.py]: @bsipocz , this okay?

Comment (bsipocz) [reply astropy/stats/biweight.py]: As far as I know yes.

**Comment by pllim** [comment]
MPL failures are unrelated.

File: astropy/stats/biweight.py [hunk 2]
```diff
@@ -328,51 +347,59 @@ def biweight_midvariance(data, c=9.0, M=None, axis=None,
...
     u = d / (c * mad)
 
     # now remove the outlier points
-    mask = np.abs(u) < 1
+    # ignore RuntimeWarnings for comparisons with NaN data values
+    with np.errstate(invalid='ignore'):
+        mask = np.abs(u) < 1
     u = u ** 2
 
+    nanmask = np.ones(data.shape)
```
Comment (bsipocz) [inline astropy/stats/biweight.py]: nitpicking, but this var name is a bit confusing as otherwise we use "mask" the other way around.

**Comment by larrybradley** [comment]
CI is all agree and approved.  Merging now.

**Comment by bsipocz** [comment]
Haha, @larrybradley, you've been just a tiny bit quicker than I 😄

File: astropy/stats/biweight.py (hunk 2 above)
Comment (larrybradley) [reply astropy/stats/biweight.py]: Yeah I know, but then I would have to invert the mask to use it.  This prevents an unnecessary computation.  I'm working on a followup PR to allowed masked arrays that will change this line.  Name suggestions for an inverse mask?

Comment (pllim) [reply astropy/stats/biweight.py]: `inversed_mask`? :sweat_smile:

Comment (pllim) [reply astropy/stats/biweight.py]: `inverted_mask`? `son_of_the_mask`?

Comment (bsipocz) [reply astropy/stats/biweight.py]: yes, I don't suggest to invert it, it was more like a naming issue. But then again, the code after that is trivial, it's not one of those cases where the variable was used again hundreds of  lines later.

Comment (pllim) [reply astropy/stats/biweight.py]: I know you're talking about cone search, Brigitta! :joy_cat:

Comment (bsipocz) [reply astropy/stats/biweight.py]: I wasn't even, honestly I haven't looked into that code at all, but now I know to stay away 😺

**Pull request merged**